### PR TITLE
refactor(databases): Creation of the databases folder to keep the logic for sqlite and postgres

### PR DESCRIPTION
**Pull request by Ivansete-status**
# Description
Simple refactoring to move all the database logic in one place (common/databases)

### CODE REVIEW

File: waku/common/databases/db_sqlite.nim [hunk 1]
```diff
@@ -345,3 +345,141 @@ proc setUserVersion*(database: SqliteDatabase, version: int64): DatabaseResult[v
...
+  # Run the migration scripts
+  for script in scripts:
+
+    for statement in script.breakIntoStatements():
+      debug "executing migration statement", statement=statement
+
+      let execRes = db.query(statement, NoopRowHandler)
+      if execRes.isErr():
+        error "failed to execute migration statement", statement=statement, error=execRes.error
+        return err("failed to execute migration statement")
```
Comment (SionoiS) [inline waku/common/databases/db_sqlite.nim:464]: Question: what happen if it fail in the middle of a migration does it need to be rolled back or no?

File: waku/common/databases/db_postgres/pgasyncpool.nim [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    while pool.state == PgAsyncPoolState.Closing:
+      await sleepAsync(0.milliseconds) # Do not block the async runtime
+    return ok()
+
+  pool.state = PgAsyncPoolState.Closing
+
+  # wait for the connections to be released and close them, without
+  # blocking the async runtime
+  if pool.conns.anyIt(it.busy):
+    while pool.conns.anyIt(it.busy):
```
[the remark this answers is not in the document]
Comment (Ivansete-status) [reply waku/common/databases/db_postgres/pgasyncpool.nim:90]: I think you are absolutely right. Thanks for it! I'll check how it works but is redundant indeed.

File: waku/common/databases/db_sqlite.nim (hunk 1 above)
Comment (Ivansete-status) [reply waku/common/databases/db_sqlite.nim:464]: Good point. We are assuming that either all the query works or all the query fails but we are not considering the case where the query partially worked in which case we should make sure the database is rolled back to the initial state. However, I'd consider that optimization for further PRs. I'd like to keep this PR with minimal changes, i.e. only rename files or delete the `if` statement that you suggested :)

Comment (Ivansete-status) [reply waku/common/databases/db_sqlite.nim:464]: Btw, I just said "just rename files" but precisely at this point, I'm adding content from "waku/common/sqlite/migrations.nim", which I removed. If you are not happy with that I can restore back the 'migrations.nim'.

Comment (SionoiS) [reply waku/common/databases/db_sqlite.nim:464]: Totally fine to not consider this edge case for now.

**Review by SionoiS — APPROVED** [review]

**Review by jm-clius — APPROVED** [review]
LGTM! Thanks for your patience :)

**Pull request merged**